client/ayon_traypublisher/plugins/create/create_editorial_advanced.py

```diff
@@ -0,0 +1,1341 @@
+                    otio_clip,
+                    instance_data,
+                    track_start_frame,
+                    folder_entity
+                )
+
+                parenting_data = {
+                    "instance_label": None,
+                    "instance_id": None
+                }
+
+                self._make_shot_product_instance(
+                    otio_clip,
+                    deepcopy(base_instance_data),
+                    parenting_data,
+                )
+
+                abs_clip_folder = os.path.join(
+                    media_folder_path, otio_clip.name).replace("\\", "/")
+
+                for pres_product_data in product_type_presets:
+                    self._make_product_instance(
+                        pres_product_data,
+                        deepcopy(base_instance_data),
+                        parenting_data,
+                        clip_related_content,
+                        abs_clip_folder,
+                    )
+
+    def _include_files_for_processing(
+        self,
+        product_name,
+        files,
+        product_data_base,
+        collecting_items,
+        strict=True,
+    ):
+        """Supporting function for getting clip content.
+
+        Args:
+            product_name (str): product name
+            partial_path (str): clip folder path
+            files (list): list of files in clip folder to collect
+            product_data_base (dict): product data
+            collecting_items (list): list for collecting product data items
+            strict (Optional[bool]): strict mode for filtering files
+        """
+        # compile regex pattern for matching product name
+        pattern_search = re.compile(
+            f".*({re.escape(product_name)}{VARIANTS_PATTERN})"
+        )
+        self.log.warning(f">>> pattern_search: {pattern_search}")
+
+        # find intersection between files and sequences
+        differences = find_string_differences(files)
+        self.log.warning(f">>> files: {pformat(files)}")
+
+        collections, reminders = clique.assemble(files)
+        self.log.warning(
+            f">>> reminders: {pformat([rem for rem in reminders])}")
+
+        # iterate all collections and search for pattern in file name head
+        for collection in collections:
+            # check if collection is not empty
+            if not collection:
+                continue
+            # check if pattern in name head is present
+            head = collection.format("{head}")
+            tail = collection.format("{tail}")
+            match = pattern_search.search(head)
+
+            # if pattern is not present in file name head
+            if strict and not match:
+                continue
+
+            # add collected files to list
+            files_ = [
+                file
+                for file in files
+                if file.startswith(head)
+                if file.endswith(tail)
+                if "thumb" not in file
```

Comment:
I'm not sure how this is safe, check for `thumb` in file does not seem like good idea. All "thumbs" assets will be skipped...

Reply:
very unlikely but true. Well it is not that easy to identify single image file which is having also thumb in name to be thumbnail. Any suggestion? but really we are talking about quite rare use case.

Reply:
I would add at least some comment like `NOTE: Guess thumbnail file - potential danger`.